nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -2,34 +2,59 @@
 
 import java.net.URI;
 import java.util.List;
-import java.util.Optional;
 import no.sikt.nva.nvi.common.NviCandidateRepository;
-import no.sikt.nva.nvi.common.model.business.ApprovalStatus;
-import no.sikt.nva.nvi.common.model.business.Candidate;
-import no.sikt.nva.nvi.common.model.business.Status;
+import no.sikt.nva.nvi.common.model.dao.ApprovalStatus;
+import no.sikt.nva.nvi.common.model.dao.Candidate;
+import no.sikt.nva.nvi.common.model.dao.Level;
+import no.sikt.nva.nvi.common.model.dao.PublicationDate;
+import no.sikt.nva.nvi.common.model.dao.Status;
+import no.sikt.nva.nvi.common.model.dao.VerifiedCreator;
+import no.sikt.nva.nvi.common.model.dto.CandidateDetailsDto;
+import no.sikt.nva.nvi.common.model.dto.EvaluatedCandidateDto;
+import no.sikt.nva.nvi.common.model.dto.PublicationDateDto;
+import no.sikt.nva.nvi.common.model.dto.VerifiedCreatorDto;
+import nva.commons.core.JacocoGenerated;
 
 public class NviService {
 
+    public static final String NVI_CANDIDATE = "Candidate";
     private final NviCandidateRepository nviCandidateRepository;
 
     public NviService(NviCandidateRepository nviCandidateRepository) {
         this.nviCandidateRepository = nviCandidateRepository;
     }
 
-    public Optional<Candidate> getCandidateByPublicationId(URI publicationId) {
-        return nviCandidateRepository.findByPublicationId(publicationId.toString());
+    //TODO: Remove JacocoGenerated when other if/else cases are implemented
+    @JacocoGenerated
+    public void upsertCandidate(EvaluatedCandidateDto evaluatedCandidate) {
+        if (isNotExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
+            createCandidate(evaluatedCandidate);
+        }
     }
 
-    public boolean exists(URI publicationId) {
-        return nviCandidateRepository.findByPublicationId(publicationId.toString()).isPresent();
+    private static List<VerifiedCreator> mapToVerifiedCreators(List<VerifiedCreatorDto> verifiedCreatorDtos) {
+        return verifiedCreatorDtos.stream()
+                   .map(verifiedCreatorDto -> new VerifiedCreator(
+                       verifiedCreatorDto.id(), verifiedCreatorDto.nviInstitutions()))
+                   .toList();
```

Comment:
you can create method toDao() for your dtos, for example candidate.toDao()

Reply:
The same for dao, candidateDao.toDto() or toCandidate(), I guess sondre will need these methods